Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1726,6 +1732,57 @@ namespace Tensile
         return size;
     }
 
+    float lerp(float x, float x0, float x1, float y0, float y1)
+    {
+        return y0 + (x - x0) * (y1 - y0) / (x1 - x0);
+    }
+
+    int ContractionSolution::predictNumFullTiles(Problem const&  problem,
+                                                 Hardware const& hardware,
+                                                 size_t          tiles,
+                                                 size_t          skGrid) const
+    {
+        int itersPerTile = problem.getItersPerTile(sizeMapping);
+        int k            = itersPerTile * sizeMapping.depthU;
+        // TODO Test if prediction should be normalized to iterations (DepthU) or full size
+
+        auto remainder = tiles % skGrid;
+        if(remainder == 0)
+            return 0;
+
+        // TODO Generalize code to allow prediction models for additional architectures
```

Comment:
It might be better to return 0 or 1 (not sure which is more appropriate) if cuCount is different from expectation?

Reply:
That's a good idea! Just pushed a change to add it.